src/app/assessment/assessment.component.ts

```diff
@@ -455,11 +455,32 @@ export class AssessmentComponent extends RouterEnter {
   /**
    * handle submission and autosave
    * @param {boolean} saveInProgress set true for autosaving or it treat the action as final submision
+   * @param {boolean} goBack use to unlock team assessment when leave assessment by clicking back button
+   * @param {boolean} isManualSave use to detect manual progress save
    */
-  async submit(saveInProgress: boolean, goBack?: boolean): Promise<any> {
+  async submit(saveInProgress: boolean, goBack?: boolean, isManualSave?: boolean): Promise<any> {
+
+    /**
+     * checking is this a submission or progress save
+     * - if it's a submission
+     *    - assign false to saving variable to disable save
+     *    - changing submitting variable value to 'Submitting'
+     * - if it's a progress save
+     *    - if this is a manual save or there are no any auto save in progress
+     *      - change saving variable value to true to enable save
+     *      - make manual save button disable
+     *      - change savingMessage variable value to 'Saving...' to show save in progress
+     *    - if this not manual save or there is one save in progress
+     *      - do nothing
+     */
     if (saveInProgress) {
-      this.savingMessage = 'Saving...';
-      this.savingButtonDisabled = true;
+      if (isManualSave || !this.saving) {
+        this.savingMessage = 'Saving...';
+        this.saving = true;
+        this.savingButtonDisabled = true;
+      } else {
+        return;
```

Comment:
If `isManualSave = false` , it will return directly here. 
That means we will never trigger auto save?